temporal/fx.go

```diff
@@ -733,10 +733,16 @@ func loadClusterInformationFromStore(ctx context.Context, config *config.Config,
 			return err
 		}
 		metadata := item.(*persistence.GetClusterMetadataResponse)
+		shardCount := metadata.HistoryShardCount
+		if shardCount == 0 {
+			// This is to add backward compatibility to the config based cluster connection.
```

Comment:
Do I understand correct this only applies to ancient versions of Temporal Server?

Reply:
Recent versions should always have this stored in DB?

Reply:
Yes.

Reply:
Yes. If the clusters are using tctl to connect, then it should have the shard count. If the clusters connection is the legacy config-base, the shard count could be 0. We don't want to support the config base connection so I just assume the clusters shard count must be equal if they are using config base connection.